### PR TITLE
[BugFix] Fix bug for binary_cross_entropy_with_logits loss

### PR DESCRIPTION
**Pull request by DesmonDay**
### PR types
Bug fixes

### PR changes
OPs

### Description
Fix bug of pos_weight calculation for binary_cross_entropy_with_logits loss. Related issue: https://github.com/PaddlePaddle/Paddle/issues/54730

截屏2023-06-26 下午3 38 25

### CODE REVIEW

**Review by GuoxiaWang — APPROVED** [review]
LGTM

File: paddle/phi/api/yaml/op_compat.yaml [hunk 1]
```diff
@@ -2800,7 +2800,7 @@
 - op: sigmoid_cross_entropy_with_logits
   backward: sigmoid_cross_entropy_with_logits_grad
   inputs :
-    {x: X, label: Label}
+    {x: X, label: Label, pos_weight: PosWeight}
```
Comment (chenwhql) [inline paddle/phi/api/yaml/op_compat.yaml]: 这个不用改，这是为了兼容原先的写法的，新增参数，不需要兼容旧的，后续新IR重构后原先驼峰式的命名写法都会删除

File: python/paddle/nn/functional/loss.py [hunk 2]
```diff
@@ -847,10 +843,13 @@ def binary_cross_entropy_with_logits(
...
                 paddle.multiply(label, paddle.subtract(pos_weight, one)), one
             )
-            pos_weight_name = (
-                name if reduction == 'none' and weight is None else None
-            )
-            out = paddle.multiply(out, log_weight, name=pos_weight_name)
+
+        helper.append_op(
+            type="sigmoid_cross_entropy_with_logits",
+            inputs={"X": logit, "Label": label, "PosWeight": log_weight},
```
Comment (chenwhql) [inline python/paddle/nn/functional/loss.py]: 这里用pos_weight即可

File: paddle/phi/infermeta/multiary.cc [hunk 3]
```diff
@@ -3410,5 +3410,61 @@ void WeightedSampleNeighborsInferMeta(const MetaTensor& row,
   out_count->set_dims({-1});
   out_count->set_dtype(DataType::INT32);
 }
+
+void SigmoidCrossEntropyWithLogitsInferMeta(const MetaTensor& x,
```
Comment (zyfncg) [inline paddle/phi/infermeta/multiary.cc]: 函数位置按字典序放置

File: paddle/phi/infermeta/multiary.h [hunk 4]
```diff
@@ -646,4 +646,12 @@ void MoeInferMeta(const MetaTensor& x,
                   const std::string& act_type,
                   MetaTensor* out);
 
+void SigmoidCrossEntropyWithLogitsInferMeta(const MetaTensor& x,
```
Comment (zyfncg) [inline paddle/phi/infermeta/multiary.h]: 同上

File: paddle/phi/api/yaml/op_compat.yaml (hunk 1 above)
Comment (DesmonDay) [reply paddle/phi/api/yaml/op_compat.yaml]: done

File: python/paddle/nn/functional/loss.py (hunk 2 above)
Comment (DesmonDay) [reply python/paddle/nn/functional/loss.py]: done

File: paddle/phi/infermeta/multiary.cc (hunk 3 above)
Comment (DesmonDay) [reply paddle/phi/infermeta/multiary.cc]: done

File: paddle/phi/infermeta/multiary.h (hunk 4 above)
Comment (DesmonDay) [reply paddle/phi/infermeta/multiary.h]: done

**Review by chenwhql — APPROVED** [review]

**Review by Shixiaowei02 — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]
